src/RevOpts.cc

```diff
@@ -49,226 +26,117 @@ void RevOpts::SetArgs( const SST::Params& params ) {
   }
 }
 
-bool RevOpts::InitPrefetchDepth( const std::vector<std::string>& Depths ) {
-  std::vector<std::string> vstr;
-  for( uint32_t i = 0; i < Depths.size(); i++ ) {
-    std::string s = Depths[i];
+template<typename MAP>
+bool RevOpts::InitPropertyMap( const std::vector<std::string>& Opts, MAP& map ) {
+  for( auto& s : Opts ) {
+    std::vector<std::string> vstr;
+
     splitStr( s, ":", vstr );
     if( vstr.size() != 2 )
       return false;
 
-    uint32_t Core = std::stoul( vstr[0], nullptr, 0 );
-    if( Core > numCores )
+    auto Core = std::stoull( vstr[0], nullptr, 0 );
+    if( Core >= numCores )
       return false;
 
-    std::string::size_type sz          = 0;
-    uint32_t               Depth       = std::stoul( vstr[1], &sz, 0 );
+    // Store as cast integer if target is integer; otherwise store as string
+    auto parse = [&]( auto val ) {
+      if constexpr( std::is_integral_v<decltype( val )> ) {
+        map[Core] = decltype( val )( std::stoull( vstr[1], nullptr, 0 ) );
+      } else {
+        map[Core] = vstr[1];
+      }
+    };
 
-    prefetchDepth.find( Core )->second = Depth;
-    vstr.clear();
+    if constexpr( is_vector<MAP>::value ) {
+      parse( typename MAP::value_type{} );
+    } else {
+      parse( typename MAP::mapped_type{} );
+    }
```

Comment:
Here the `parse` lambda defined above is called one of two ways based on whether `MAP` is a vector type or not. If it's a vector, then its `value_type` member defines the vector element type, and we pass a default `{}` (or zero) value of it. If it's not a vector, we assume it's a `map` / `unordered_map` and get the value type from its `mapped_type` member and pass a similar value to the lambda.